modin/backends/base/query_compiler.py

```diff
@@ -892,7 +892,23 @@ def fillna(self, **kwargs):
         Returns:
             A new QueryCompiler with null values filled.
         """
-        return DataFrameDefault.register(pandas.DataFrame.fillna)(self, **kwargs)
+
+        def fillna(df, value, **kwargs):
+            if (
+                isinstance(df, pandas.DataFrame)
+                and len(df.columns) == 1
+                and "__reduced__" in df.columns
+            ):
+                df = df.squeeze(axis=1)
+            if (
+                isinstance(value, pandas.DataFrame)
+                and len(value.columns) == 1
+                and "__reduced__" in value.columns
+            ):
+                value = value.squeeze(axis=1)
```

Comment:
if `value` was originally Series, and its name was not None, then its column contains Series name, not `"__reduced__"`

Reply:
Thank you for the information, but it means that there is no nearly reliable way to distinguish a single column DataFrame from a Series type.
I am out of ideas of how to implement this place, so any suggestions are welcome.

Reply:
Type information is now passed by means of two additional parameters.